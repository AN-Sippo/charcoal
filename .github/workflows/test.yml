name: test
on:
  push:
    branches:
      - main

  pull_request:
    branches:
      - main

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version-file: '.node-version'
          cache: yarn

      - name: Prepare deps
        run: |
          yarn install --immutable --inline-builds
          yarn build

      # TODO: jobを分けてビルド結果をキャッシュして並列化
      - name: Test
        run: |
          yarn test
          yarn lint
          yarn typecheck:config
          yarn typecheck

      - name: Check if Storybook builds
        # storybook の vite build が OOM で落ちることがある
        # これだけ違う job に分離して強い runner で動かすと良いかも？
        run: |
          yarn build-storybook
          NODE_OPTIONS=--max_old_space_size=6144 USE_VITE=1 yarn build-storybook
