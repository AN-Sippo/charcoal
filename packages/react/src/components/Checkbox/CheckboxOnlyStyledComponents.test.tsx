import 'jest-styled-components'

import React from 'react'
import CheckboxOnlyStyled from './CheckboxOnlyStyledComponents'
import { render } from './performance.test'

describe('CheckboxOnlyStyled', () => {
  test('<CheckboxOnlyStyled />', () => {
    expect(render(<CheckboxOnlyStyled label="test" />)).toMatchSnapshot()
  })
})
