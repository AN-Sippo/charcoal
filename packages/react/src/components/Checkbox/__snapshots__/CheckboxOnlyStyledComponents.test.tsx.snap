// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxOnlyStyled <CheckboxOnlyStyled /> 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  gap: 4px;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
  opacity: 0.32;
}

.c1 {
  position: relative;
}

.c2[type='checkbox'] {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  cursor: pointer;
  margin: 0;
  width: 20px;
  height: 20px;
  -webkit-transition: 0.2s box-shadow;
  transition: 0.2s box-shadow;
  border-radius: 4px;
}

.c2[type='checkbox']:checked {
  background: #0096fa;
}

.c2[type='checkbox']:not(:checked) {
  border-width: 2px;
  border-style: solid;
  border-color: #adadad;
}

.c2[type='checkbox']:focus {
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c3 {
  position: absolute;
  top: -2px;
  left: -2px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
  color: #ffffff;
  visibility: hidden;
}

<label
  aria-disabled={false}
  className="c0"
>
  <div
    className="c1"
  >
    <input
      aria-label="test"
      checked={false}
      className="c2"
      type="checkbox"
    />
    <div
      aria-hidden={true}
      checked={false}
      className="c3"
    >
      <pixiv-icon
        name="24/Check"
        unsafe-non-guideline-scale={0.6666666666666666}
      />
    </div>
  </div>
</label>
`;
